From 5b913bd715be73df8a931522cb31c36fc9f53f33 Mon Sep 17 00:00:00 2001
From: Frode Nordahl <fnordahl@ubuntu.com>
Date: Thu, 5 Sep 2024 13:58:44 +0200
Subject: [PATCH] northd: Dont block RAs from BGP redirect port.

From: Martin Kalcok <martin.kalcok@canonical.com>

BGP unnumbered relies on IPv6 RAs to form peer connections,
however unsolicited RAs from Logical Router do no seem to
reach external networks [0]. Therefore we need to allow
unsolicited RAs from the port designated for BGP redirect.

[0]https://mail.openvswitch.org/pipermail/ovs-discuss/2024-August/053282.html

Signed-off-by: Martin Kalcok <martin.kalcok@canonical.com>
---
 northd/northd.c | 7 -------
 1 file changed, 7 deletions(-)

diff --git a/northd/northd.c b/northd/northd.c
index 9fd225d77..bd6deb245 100644
--- a/northd/northd.c
+++ b/northd/northd.c
@@ -14754,13 +14754,6 @@ build_lrouter_routing_protocol_redirect(
                   REGBIT_PORT_SEC_DROP " = 1; next;",
                   lflow_ref);
 
-    ds_clear(match);
-    ds_put_format(match, "inport == \"%s\" && nd_ra",
-                  redirect_port_name);
-    ovn_lflow_add(lflows, op->peer->od, S_SWITCH_IN_CHECK_PORT_SEC, 80,
-                  ds_cstr(match),
-                  REGBIT_PORT_SEC_DROP " = 1; next;",
-                  lflow_ref);
     free(redirect_port_name);
 }
 
-- 
2.45.2

